library/src/main/java/org/jellyfin/apiclient/discovery/DiscoveryService.kt

```diff
@@ -0,0 +1,41 @@
+package org.jellyfin.apiclient.discovery
+
+/**
+ * Service for discovery related functionality
+ */
+class DiscoveryService(
+	private val discoveryBroadcastAddressesProvider: DiscoveryBroadcastAddressesProvider
+) {
+	private val localServerDiscovery by lazy {
+		// Create instance
+		LocalServerDiscovery(discoveryBroadcastAddressesProvider)
+	}
+
+	/**
+	 * Parses the given [input] and tries to fix common mistakes.
+	 *
+	 *   - Add protocol (https and http)
+	 *   - Add port (8096 for http and 8920 for https)
+	 *   - Add baseurl (/jellyfin)
+	 *
+	 * The returned candidates are in order of most secure followed by most likely to work.
+	 * See [AddressCandidateHelper] for more information.
+	 */
+	fun getAddressCandidates(input: String) = AddressCandidateHelper(input).run {
+		addCommonCandidates()
+		prioritize()
```

Comment:
Maybe put this into the `getCandidates` function even?

Reply:
I explicitly did no do that if an unsorted list is desired.

Reply:
But I added an additional comment to the `getCandidates` function

Reply:
> I explicitly did no do that if an unsorted list is desired.

Hmm, I see. You have a typo in there though, I think.

Reply:
You're right. I shouldn't work on 5 things at once lol